syndesis-maven-plugin/src/main/java/io/syndesis/maven/GenerateMapperInspectionsMojo.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.maven;
+
+import io.atlasmap.maven.GenerateInspectionsMojo;
+import io.syndesis.dao.init.ModelData;
+import io.syndesis.dao.init.ReadApiClientData;
+import io.syndesis.model.Kind;
+import io.syndesis.model.connection.Action;
+import io.syndesis.model.connection.Connector;
+import io.syndesis.model.connection.DataShape;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+
+@Mojo(name = "generate-mapper-inspections", defaultPhase = LifecyclePhase.GENERATE_RESOURCES)
+public class GenerateMapperInspectionsMojo extends AbstractMojo {
+
+    @Parameter(readonly = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    @Component
+    private RepositorySystem system;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true)
+    private List<RemoteRepository> remoteRepos;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+    private RepositorySystemSession repoSession;
+
+    @Parameter(defaultValue = "${project.build.directory}/generated-sources/atlasmap")
+    private File outputDir;
+
+    @Parameter(defaultValue = "static/mapper/v1/java-inspections")
+    private String resourceDir;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        try {
+
+            Resource resource = new Resource();
+            resource.setDirectory(outputDir.getCanonicalPath());
+            project.addResource(resource);
+
+            HashSet<File> generated = new HashSet<>();
+
+            ReadApiClientData reader = new ReadApiClientData();
+            List<ModelData<?>> modelList = reader.readDataFromFile("io/syndesis/dao/deployment.json");
```

Comment:
wouldn't it make sense to provide an explicit list to configure for the data types to extract ? And that already in the upstream plugin. 

I'm not sure that extracting the shapes from demo data is sufficient and reliable. Would prefer something like:

```xml
<configuration>
   <types>
       <twitter>
            <type>twitterj.Status</type>
             <type>...</type>
        </twitter>
        <saleforce>
             .....
        </salesforce>
   </types>
</configuration>
```

It's a bit tricky to setup this kind of configuration for Maven (but it works when using sth. like `Map<String, ArrayList>` as type for a `types` configuration field, see also [here](http://stackoverflow.com/questions/38628399/using-map-of-maps-as-maven-plugin-parameters/38642613) , but I could help after my PTO (next week). 

In the meantime, I'm ok with this as a temporary solution (but please open issue for the refactoring if you agree that we should couple the generation so tightly to the content of a file which can change anytime, or even vanish)